pysd/py_backend/vensim/vensim2py.py

```diff
@@ -350,6 +367,50 @@ def get_external_data(func_str, args_str, root_path):
 
     return f(*args, root=root_path).subscript
 
+def get_subscript_number_range(subs_start, subs_end):
+    """
+    With the first and the last subscript values of a subscript
+    numeric range, gets the common string of both and the 
+    starting and ending number of the numeric range
+
+    Parameters
+    ----------
+    subs_start: str
+        Represents the first subscript value in the numeric range
+    subs_end: str
+        Represents the last subscript value in the numeric range
+    
+    Returns
+    -------
+    common: str
+        Common string of both subscripts
+    num_start: int
+        Sequence start number 
+    num_end: int
+        Sequence end number
+    
+    Examples
+    --------
+    >>> get_subscript_number_range('Layer1', 'Layer5')
+    ('Layer', 1, 5)
+    >>> get_subscript_number_range('sub15', 'sub30')
+    ('sub', 15, 30)
+    """
+    subs_start_l = list(subs_start)
+    subs_end_l = list(subs_end)
+    common = ""
+    for i in range(len(subs_start)):
+        if(subs_start_l[i] == subs_end_l[i] and not(subs_start_l[i].isdigit())):
+            common = common + subs_start_l[i]
+        else: break
```

Comment:
This may crash for a range defined as `(Y2020M1-Y2020M12)`.
Using regex as shown below would prevent that:
```
subs_start = re.findall('\d+|\D+', subs_start)
subs_end = re.findall('\d+|\D+', subs_end)
prefix = ''.join(subs_start[:-1])
num_start = int(subs_start[-1])
num_end = int(subs_end[-1])
```

Reply:
Yes, good point, thank you! 
I have changed the grammar and the function and now it works in that case.

Reply:
Perfect, you're welcome :)